src/components/LibraryDetailPage.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import { Store } from "redux";
+import { State } from "../reducers/index";
+import LibraryDetailItem from "./LibraryDetailItem";
+import LibraryStageItem from "./LibraryStageItem";
+import SaveButton from "./SaveButton";
+import Form from "./Form";
+
+export interface LibraryDetailPageDispatchProps {
+  editStages: (data: FormData) => Promise<void>;
+}
+
+export interface LibraryDetailPageStateProps {
+  fullLibrary?: LibraryData;
+}
+
+export interface LibraryDetailPageOwnProps {
+  library: LibraryData;
+  store: Store<State>;
+  uuid: string;
+  updateColor: (library_stage: string, registry_stage: string) => void;
+}
+
+export interface LibraryDetailPageState {
+  libraryStage: string;
+  registryStage: string;
+}
+
+export interface LibraryDetailPageProps extends LibraryDetailPageStateProps, LibraryDetailPageDispatchProps, LibraryDetailPageOwnProps {}
+
+export class LibraryDetailPage extends React.Component<LibraryDetailPageProps, LibraryDetailPageState> {
+
+  constructor(props: LibraryDetailPageProps) {
+    super(props);
+    this.submit = this.submit.bind(this);
+    this.renderInfo = this.renderInfo.bind(this);
+    this.renderStages = this.renderStages.bind(this);
+    this.state = { libraryStage: this.props.library.library_stage, registryStage: this.props.library.registry_stage };
+  }
+
+  renderInfo() {
+    let fields = Object.keys(this.props.library).map(label =>
+      <LibraryDetailItem key={label} label={label} value={this.props.library[label]} />
+    );
+    return fields;
+  }
+
+  renderStages() {
+    return (
+      <Form
+        hiddenName="uuid"
+        hiddenValue={this.props.uuid}
+        onSave={this.submit}
+        content={[
+          <LibraryStageItem key="lib" label={"Library Stage"} value={this.state.libraryStage} />,
+          <LibraryStageItem key="reg" label={"Registry Stage"} value={this.state.registryStage} />
+        ]}
+      />
+    );
+  }
+
+  async submit(data): Promise<void> {
+    this.props.updateColor(data.get("Library Stage"), data.get("Registry Stage"));
+    await this.props.editStages(data);
```

Comment:
It's possible this could fail, no? If it does, you'd be updating the color to the right value but because the server couldn't update the values, they wouldn't be right.

Reply:
Yes.  But I haven't done anything with error handling anywhere, yet; I figure that's in the next phase...